src/resources/MachineLearning/ModelDetailedInfo/details/ModelDetails.ts

```diff
@@ -13,7 +13,7 @@ export interface ModelDetails {
     languages?: Map<string, ModelDetailsLanguages>;
     candidateExamples?: Map<string, string[]>;
     candidatesPerLanguages?: Map<string, number>;
-    minClickCountPerLang?: Map<string, number>;
+    minClickCountByLang?: Map<string, number>;
```

Comment:
Are you sure all of those are Map?

https://www.typescriptlang.org/play/?#code/C4TwDgpgBAglC8UCyBDMAeAzsATgSwDsBzAGigIFcBbAIwhwD4AoUSKAIQSgCUIBjAPY4AJllyFS5anUZM5ggtigoAXLC4BvAEQotagIwBfANxMFSmms6JtNPVCOn5AxcCg4IKADYw1cRAQQAO7IaAAUANoROlpk+gC68QCUph7eMAB0ABYomGExSUA

Reply:
https://graphite.dev/guides/typescript-record-utility-type#comparing-object-object-with-object-object-in-typescript, both would work however we have dynamic keys in this case which would make Map a better choice performance-wise according to that article.
Technically our keys here are scoped down to language codes and unlimited (well as many languages as coveo wants to support) so the scope becomes quite large and is quite unpredictable. We actually access the values from a list of keys provided in another property as well which is why we dig into them like this.{modelDetails?.candidatesPerLanguages?.[language] || 0}

If you see an advantage to switching over we could open a ticket and do it another PR.

Reply:
There are benefits of using a `Map` vs using a `Record` (plain object), but the JSON the backend returns cannot contain a `Map` except if you're manually doing the conversion when receiving the data, which [is not done](https://github.com/coveo/platform-client/blob/99b5bbb304c755091d58f2311ab26b56ae7e7329/src/resources/MachineLearning/ModelDetailedInfo/ModelDetailedInfo.ts#L9). My point here is that you're using the wrong type and "lie" to TypeScript about the data type we receives from the backend

With a `Map` you cannot use brackets to get the value, but you can using a `Record`

<img width="792" alt="image" src="https://github.com/user-attachments/assets/5f44f7e2-fa90-443a-9781-ef716ee7b798" />

Map has 

>No native support for serialization or parsing.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#objects_vs._maps

Reply:
sounds good TIL, https://coveord.atlassian.net/browse/MLX-1807 I've created a ticket since the scope of this task is to rename the property to what is expected. If we do the change in a separate PR it'll allow us to do the change and reduce the scope of testing for each change.
https://coveord.atlassian.net/browse/MLX-1807